### PR TITLE
ispc 1.10.0: fix standard library generation and add meaningful test

### PR DESCRIPTION
**Pull request by jamiesnape**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

#38637 is failing because the ispc standard library is not available. Closer inspection of the build of this formula shows a silent failure (other than console output when compiled verbosely). This is specific to the use of the `llvm@4` formula as a dependency of this formula and not necessarily an upstream bug (though the silent failure is not very helpful). Note also https://github.com/jvo203/fits_web_ql/commit/62ba2e29fc0676fca6315d0021581bba38476cd2#diff-04c6e90faac2675aa89e2176d2eec7d8R73. Also adds a test that would have failed without this fix.

### CODE REVIEW

**Comment by jamiesnape** [comment]
(To be clear, the formula is currently broken without this.)

File: Formula/ispc.rb [hunk 1]
```diff
@@ -17,9 +18,16 @@ class Ispc < Formula
...
+    # The standard include paths for clang supplied by the llvm@4 formula do not include
+    # C headers such as unistd.h. Add the path to those headers explicitly so that
+    # generation of the ispc builtins and standard library do not silently fail.
+    inreplace "cmake/GenerateBuiltins.cmake", "${CLANG_EXECUTABLE}",
+      "${CLANG_EXECUTABLE} -I#{MacOS.sdk_path}/usr/include"
+
     args = std_cmake_args + %W[
       -DISPC_INCLUDE_EXAMPLES=OFF
       -DISPC_INCLUDE_TESTS=OFF
+      -DISPC_INCLUDE_UTILS=OFF
```
Comment (zbeekman) [inline Formula/ispc.rb]: Seems orthogonal to the issue. What motivated this change?

Comment (jamiesnape) [reply Formula/ispc.rb]: Parity with the previous version of the formula, i.e., an omission when the formula switched build systems.

**Review by zbeekman — APPROVED** [review]
LGTM

Thanks for the better tests!

**Comment by zbeekman** [comment]
Something went wrong with the bottles (sha256 missmatch) so I re-triggered the build.

**Comment by zbeekman** [comment]
@BrewTestBot test this please
